src/app.ts

```diff
@@ -35,23 +47,161 @@ const app = new Koa();
+  .get("/api/operations", async ctx => {
+    // TODO: Do we need to paginate?
+    const ops = await operations.orderBy("created_at").get();
+    const parsedOps: Array<Object> = [];
+    ops.forEach(op =>
+      parsedOps.push({
+        id: op.id,
+        creator_mid: op.get("creator_mid"),
+        creator_uid: op.get("creator_uid"),
+        op_code: op.get("op_code"),
+        data: op.get("data")
+      })
+    );
+    ctx.body = JSON.stringify(parsedOps);
+    ctx.status = 200;
+    return;
+  })
+  /* These endpoints are consumed by coconut. */
+  .post("/api/members/:uid/notify_video_encoded", async ctx => {
+    try {
+      const videoRef = getPrivateVideoRef(ctx.state.toMember.id);
+      if ((await videoRef.exists())[0]) {
+        await videoRef.delete();
+      } else {
+        console.warn("We expected a private video file to exist that did not.");
+      }
+      ctx.status = 200;
+    } catch (error) {
+      console.error(error);
+      ctx.status = 500;
+    }
+    return;
+  })
+  .post("/api/members/:uid/upload_video", ctx => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const { mid } = ctx.query;
+        if (!mid) {
+          ctx.status = 400;
+          ctx.body = "Must supply MID when uploading video.";
+          return resolve();
+        }
+
+        const videoRef = getPublicVideoRef(decodeURIComponent(mid));
+        if ((await videoRef.exists())[0]) {
+          ctx.status = 400;
+          ctx.body =
+            "Video already exists at intended storage destination. Cannot overwrite.";
+          return resolve();
+        }
+
+        const reqType = ctx.request.type;
+        if (reqType !== "multipart/form-data") {
+          ctx.status = 400;
+          ctx.body = "Must submit data as multipart/form-data";
+          return resolve();
+        }
+
+        const { files } = await asyncBusboy(ctx.req);
+        const encodedVideo = files.filter(
+          file => file.fieldname === "encoded_video"
+        );
+        if (encodedVideo.length !== 1) {
+          ctx.status = 400;
+          ctx.body = "Zero or multiple encoded videos supplied with request.";
+          return resolve();
+        }
+
+        encodedVideo[0]
+          .pipe(videoRef.createWriteStream())
+          .on("error", error => {
+            const errorMessage = "Failed to write file to Google storage.";
+            console.error(errorMessage);
+            console.error(error);
+            ctx.body = errorMessage;
+            ctx.status = 500;
+            resolve();
+          })
+          .on("finish", () => {
+            ctx.status = 201;
+            resolve();
+          });
+      } catch (error) {
+        const errorMessage =
+          "An error occurred while saving an encoded video from Coconut.";
+        console.error(errorMessage);
+        console.error(error);
+        ctx.body = errorMessage;
+        ctx.status = 500;
+        resolve();
+      }
+      return;
```

Comment:
Return statement doesn't do anything, and if this branch can be reached, the web server will just stall until it times out